compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/kotlin/KSClassDeclarationImpl.kt

```diff
@@ -52,6 +54,10 @@ class KSClassDeclarationImpl private constructor(val ktClassOrObject: KtClassOrO
         (ktClassOrObject is KtObjectDeclaration) && (ktClassOrObject.isCompanion())
     }
 
+    override fun getSealedSubclasses(): Sequence<KSClassDeclaration> {
+        return descriptor.sealedSubclassesSequence()
```

Comment:
Assuming that 99% of the classes are not sealed, can we check the modifier to return early without having to resolve to descriptor? e.g.,
```
return if (ktClassOrObject.hasModifier(KtTokens.SEALED_KEYWORD) {
    descriptor.sealedSubclassesSequence()
} else {
    emptySequence()
}
```